src/segments/pulumi.go

```diff
@@ -0,0 +1,237 @@
+package segments
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"gopkg.in/yaml.v3"
+	"time"
+)
+
+const (
+	PulumiUser properties.Property = "pulumi_user"
+	PulumiURL  properties.Property = "pulumi_url"
+)
+
+type Pulumi struct {
+	props properties.Properties
+	env   platform.Environment
+	Stack string
+
+	SpecFile
+	Backend
+}
+
+type SpecFile struct {
+	FileName string
+	Kind     string
+	Path     string
+}
+
+type Backend struct {
+	URL  string
+	User string
+}
+
+type PulumiStackLSJson struct {
+	Name          string `json:"name"`
+	Current       bool   `json:"current"`
+	ResourceCount int    `json:"resourceCount"`
+	LastUpdate    string `json:"lastUpdate"`
+}
+
+type PulumiAboutJS struct {
+	Backend *PulumiBackendJS `json:"backend"`
+}
+
+type PulumiBackendJS struct {
+	URL  string `json:"url"`
+	User string `json:"user"`
+}
+
+type PulumiFileSpec struct {
+	Name string `yaml:"name" json:"name"`
+}
+
+type PulumiWorkSpaceFileSpec struct {
+	Stack string `json:"stack"`
+}
+
+func (p *Pulumi) Template() string {
+	return "{{ .Stack }}{{if .User }} :: {{ .User }}@{{ end }}{{ if .URL }}{{ .URL }}{{ end }}"
+}
+
+func (p *Pulumi) Init(props properties.Properties, env platform.Environment) {
+	p.props = props
+	p.env = env
+}
+
+func (p *Pulumi) Enabled() bool {
+	err := p.getPulumiFile()
+	if err != nil {
+		p.env.Error(err)
+		return false
+	}
+	if p.props.GetBool(PulumiUser, false) && p.props.GetBool(PulumiURL, false) {
+		return p.getPulumiStackName() && p.getPulumiAbout()
+
+	} else {
+		return p.getPulumiStackName()
+	}
+}
+
+func (p *Pulumi) getPulumiStackName() bool {
+
+	project, err := p.getProjectName()
+	if err != nil {
+		p.env.Error(err)
+		return false
+	}
+
+	sha1HexString, err := p.sha1HexString(p.SpecFile.Path)
```

Comment:
```suggestion
	sha1HexString, err := p.sha1HexString(p.SpecFile.Path + "/" + p.SpecFile.FileName)
```

This path should include the filename